### PR TITLE
Improving indexer connector log messages

### LINKED ISSUE #22532: Worker fails to initialize IndexerConnector for index 'wazuh-states-vulnerabilities'
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.8.0-beta4 | Vulnerability Detection | Manager | Packages | AlmaLinux 9 |

### Description

During https://github.com/wazuh/wazuh/issues/22452 it has been detected that worker nodes after a few minutes, fails to initialize IndexerConnector for index 'wazuh-states-vulnerabilities', leading to a multiple error messages in the logs:

```
2024/03/14 19:06:57 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:07:57 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:08:57 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:09:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:10:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:11:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:12:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:13:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:14:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:15:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:16:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:17:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:18:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:19:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:20:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:21:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:22:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:23:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:24:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:25:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
2024/03/14 19:26:58 indexer-connector: WARNING: Error initializing IndexerConnector for index 'wazuh-states-vulnerabilities': No available server. Retrying in 60 seconds. Maximum wait time: 60 seconds.
```

This occur, even when the indexers are up and running correctly, and the worker have access to them
```
[root@almalinux4 vagrant]# filebeat test output
elasticsearch: https://192.168.56.192:9200...
  parse url... OK
  connection...
    parse host... OK
    dns lookup... OK
    addresses: 192.168.56.192
    dial up... OK
  TLS...
    security: server's certificate chain verification is enabled
    handshake... OK
    TLS version: TLSv1.3
    dial up... OK
  talk to server... OK
  version: 7.10.2
elasticsearch: https://192.168.56.193:9200...
  parse url... OK
  connection...
    parse host... OK
    dns lookup... OK
    addresses: 192.168.56.193
    dial up... OK
  TLS...
    security: server's certificate chain verification is enabled
    handshake... OK
    TLS version: TLSv1.3
    dial up... OK
  talk to server... OK
  version: 7.10.2
```

```
[root@almalinux4 vagrant]# cat /var/ossec/etc/ossec.conf | grep host 
    <hosts>
      <host>https://192.168.56.192:9200</host>
      <host>https://192.168.56.193:9200</host> 
    </hosts>
    <ignore>/etc/hosts.deny</ignore>
    <white_list>^localhost.localdomain$</white_list>
    <name>host-deny</name>
    <executable>host-deny</executable>
    <ssl_verify_host>no</ssl_verify_host>
[root@almalinux4 vagrant]# curl -k -u admin:FfbTn14Mkrdw2lWGdfla1Kf.eJLJhF35 https://192.168.56.192:9200
{
  "name" : "node-1",
  "cluster_name" : "wazuh-cluster",
  "cluster_uuid" : "-BbsJd4ZTX6GfY9fNcXmOQ",
  "version" : {
    "number" : "7.10.2",
    "build_type" : "rpm",
    "build_hash" : "eee49cb340edc6c4d489bcd9324dda571fc8dc03",
    "build_date" : "2023-09-20T23:54:29.889267151Z",
    "build_snapshot" : false,
    "lucene_version" : "9.7.0",
    "minimum_wire_compatibility_version" : "7.10.0",
    "minimum_index_compatibility_version" : "7.0.0"
  },
  "tagline" : "The OpenSearch Project: https://opensearch.org/"
}
[root@almalinux4 vagrant]# curl -k -u admin:FfbTn14Mkrdw2lWGdfla1Kf.eJLJhF35 https://192.168.56.192:9200
{
  "name" : "node-1",
  "cluster_name" : "wazuh-cluster",
  "cluster_uuid" : "-BbsJd4ZTX6GfY9fNcXmOQ",
  "version" : {
    "number" : "7.10.2",
    "build_type" : "rpm",
    "build_hash" : "eee49cb340edc6c4d489bcd9324dda571fc8dc03",
    "build_date" : "2023-09-20T23:54:29.889267151Z",
    "build_snapshot" : false,
    "lucene_version" : "9.7.0",
    "minimum_wire_compatibility_version" : "7.10.0",
    "minimum_index_compatibility_version" : "7.0.0"
  },
  "tagline" : "The OpenSearch Project: https://opensearch.org/"
}
[root@almalinux4 vagrant]# 
```

### Steps to reproduce

- Deploy a wazuh distributed environment (2 indexer, 2 managers, and 1 dashboard)
- Deploy a manager and register it to the worker node
- Wait a few minutes and check that specified warnings have been generated

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|Closes #22532|

## Description

This PR improves the messages shown to the user when the indexer connector finds an error.
Also, some callbacks were re-organized.

### CODE REVIEW

**Review by sebasfalcone — DISMISSED** [review]
LGTM

**Review by GabrielEValenzuela — DISMISSED** [review]
Great job Mati!

**Comment by pereyra-m** [comment]
The failure of the test https://github.com/wazuh/wazuh/actions/runs/8454182489/job/23158814102?pr=22682 doesn't seem related to the changes in the PR

File: src/shared_modules/indexer_connector/src/indexerConnector.cpp [hunk 1]
```diff
@@ -200,14 +200,12 @@ IndexerConnector::IndexerConnector(
                 }
                 catch (const std::exception& e)
                 {
-                    // Improved logging message
                     logWarn(IC_NAME,
-                            "Error initializing IndexerConnector for index '%s': %s. Retrying in %ld seconds. Maximum "
-                            "wait time: %ld seconds.",
+                            "Unable to initializing IndexerConnector for index '%s': %s. Retrying in %ld "
```
Comment (Dwordcito) [inline src/shared_modules/indexer_connector/src/indexerConnector.cpp:204]: This message should not be perpetual (printed every 30 seconds non-stop).

At the Warning level, we should only print it once informing that we are going to retry continuously until successful (which has to print a non-technical log ex: Initialized with success.)

The perpetual message can be at the DEBUG level, in terms of analyzing the case.

**Review by Dwordcito — CHANGES_REQUESTED** [review]

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**